frankenphp.c

```diff
@@ -105,13 +106,24 @@ static uintptr_t frankenphp_clean_server_context() {
 
 static void frankenphp_request_reset() {
   zend_try {
+    frankenphp_server_context *ctx = SG(server_context);
+
     int i;
+    if (ctx->worker_ready) {
+      for (i = 0; i < NUM_TRACK_VARS; i++) {
+        zval_ptr_dtor(&PG(http_globals)[i]);
+      }
 
-    for (i = 0; i < NUM_TRACK_VARS; i++) {
-      zval_ptr_dtor(&PG(http_globals)[i]);
+      // Restore worker script super globals
+      for (i = 0; i < NUM_TRACK_VARS; i++) {
+        ZVAL_COPY(&PG(http_globals)[i], &ctx->worker_http_globals[i]);
```

Comment:
You may have to call `GC_DEREF(&PG(http_globals)[i])` here to prevent a memory leak. I think. 

This is because you've increased the references here +1, but in `php_has_environment()` the ref will also increase +1. (IIRC) So you need to manually decrease it -1.

Reply:
or `GC_DEREF(&ctx->worker_http_global[i])`, :D

Reply:
If that's the problem, `ZVAL_COPY_VALUE` will copy the value, without increasing the reference count. But then `memcpy` should also have worked.

Reply:
I believe it may be only needed in this if-branch.

Reply:
I "fixed" the leak but as you can see the fix is bad, there is a logic error somewhere :D